spec/01-unit/08-router_spec.lua

```diff
@@ -1753,6 +1753,28 @@ for _, flavor in ipairs({ "traditional", "traditional_compatible", "expressions"
             assert.falsy(match_t)
           end)
 
+          it("update with wrong route", function()
+            local use_case = {
+              {
+                service = service,
+                route = {
+                  id = "e8fb37f1-102d-461e-9c51-6608a6bb8101",
+                  paths = { "~/delay/(?<delay>[^\\/]+)$", },
+                  updated_at = 100,
+                },
+              },
+            }
+
+            local ok, nrouter = pcall(new_router, use_case, router)
+
+            assert.falsy(router.rebuilding)
```

Comment:
We should not use `router.rebuilding` for testing this change, as it is a internal flag that may be changed later.

This should be tested with integration tests like https://github.com/Kong/kong/blob/master/spec/02-integration/05-proxy/02-router_spec.lua#L2311 that directly observes the end result of a successful rebuild instead of the flag.

Reply:
I will try it.

Reply:
done.